slave-base/contrib/bin/run-jnlp-client

```diff
@@ -110,6 +110,12 @@ if [[ $# -gt 1 ]]; then
 
   if [ -z "${JAVA_OPTS}" ]; then
       JAVA_OPTS="-XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -Dsun.zip.disableMemoryMapping=true"
+      export JAVA_OPTS
+  fi
+
+  if [ -z "${MAVEN_OPTS}" ]; then
```

Comment:
Shouldn't this be set in the maven slave? One other point worth raising with this is that sometimes maven builds might fork regular JVMs during tests, those JVMs usually don't get the right options unless set explicitly via surefire or other means. One possible work-around is to set `_JAVA_OPTIONS` an env which the [JVM will pick up](http://hg.openjdk.java.net/jdk8u/jdk8u/hotspot/file/8015f8f73cde/src/share/vm/runtime/arguments.cpp#l2641). A drawback of this is that not all JVM flags can be set with this (e.g. turning on NativeMemoryTracking). Another is that you don't see it on the ps tree, but only via the "Picked up _JAVA_OPTIONS ..." messages.

Reply:
Yeah I'm not sure how bullet proof we'll be able to get wrt auto tuning all the maven / gradle permutations.  A simple starting point in our scripts, along with some sort of generic warning in the doc about needing to understand the process model of your usage, etc. so that you can extend images and modify as needed seems more in our sweet spot.

Thoughts @bparees @jim-minter @jerboaa ?

Reply:
I have mixed feelings about whether it should be in the maven slave or here, but i think i'm inclined to agree with @jerboaa that it should be in the maven slave.

and yes, whatever we do here needs to be accompanied by better doc/warnings about "when running maven/gradle in a slave, here are settings you need to consider.... and here's what/how we attempt to default them"